### DISCUSSION TITLE
Server Side Support

### DISCUSSION THREAD

**Discussion by blamarvt** [preactjs/wmr#398] [Q&A]
Is there support for, or will there be support for easily accessing the underlying HTTP request? I'd like to provide SSR content on authenticated requests or redirect users requesting inaccessible URLs. Currently with most frameworks (or really, all that I've found) the concept of SSR falls apart when cookies/sessions are involved.

Thanks!

**Comment by JoviDeCroock** [comment]
Hey,

Currently we have no real SSR-support within `wmr`, we can statically pre-render but dynamic server-side rendering isn't currently there. That being said we have an ongoing experiment to enable SSR https://github.com/preactjs/wmr/pull/169.

Most of the time to granularly enable SSR it would be good to have a similar structure to Next where every page is a separate serverless route, this way you can define what happens on every page, another option would be to analyze the request and define the logic on the server so you can return a Redirect rather than the pre-rendered HTML, I'll bring this up internally in the team as well.
